tests/tezos-binaries.nix

```diff
@@ -0,0 +1,139 @@
+# SPDX-FileCopyrightText: 2020 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: MPL-2.0
+{ path-to-binaries ? (import ../nix { }).binaries } @ args:
+import <nixpkgs/nixos/tests/make-test-python.nix> ({ pkgs, ... }:
+let
+  tezos-accuser = "${path-to-binaries}/tezos-accuser-006-PsCARTHA";
+  tezos-admin-client = "${path-to-binaries}/tezos-admin-client";
+  tezos-baker = "${path-to-binaries}/tezos-baker-006-PsCARTHA";
+  tezos-client = "${path-to-binaries}/tezos-client";
+  tezos-endorser = "${path-to-binaries}/tezos-endorser-006-PsCARTHA";
+  tezos-node = "${path-to-binaries}/tezos-node";
+  tezos-signer = "${path-to-binaries}/tezos-signer";
+in
+{
+  nodes.machine = { ... }: { };
+
+  testScript = ''
+    machine.succeed('export TEZOS_LOG="* -> warning"')
```

Comment:
Is it a good idea to write 100+ lines of python code inside a `.nix` file?

Reply:
that's how nixos tests work :shrug:

Reply:
IMO, it's a bit less error-prone in comparison to bash script. Also one may want to add tests that include multiple machines, and it's easy to do that using nixos test engine

Reply:
But isn't it possible to put this python code into some `test.py` and somehow import it here? I know that nix is pure and `testScript = readFile(test.py)` probably won't work here directly, but is there no way to make it work? If we put it into `test.py` we will get syntax highlighting and other benefits.

Reply:
`testScript = readFile ./test.py` would indeed work if it wasn't for string interpolation. `${tezos-node}` only works in a string.

Note however that this can be circumvented in various ways.

Reply:
that would be nice, but it seems that the test engine does some formatting stuff and cannot parse scripts given as a filepaths instead of plain code

Reply:
also, something tricky should be done with binaries paths

Reply:
You can do something like this:
```
testScript = ''
tezos_node = "${tezos-node}"
tezos_client = "${tezos-client}"
<...>
${readFile ./test.py}
''
```

Reply:
:pogchamp: